### DISCUSSION TITLE
Can't get TIFF compression to work

### DISCUSSION THREAD

**Discussion by Tom-EllisEVENTS** [dlemstra/Magick.NET#1133] [Help] [answered]
I'm trying to convert PDF to multi-page TIFF and can't get the compression working. 

Currently, my code looks like this:

        using (var images = new MagickImageCollection())
        {
            // Add all the pages of the pdf file to the collection
            images.Read(e.FullPath, settings);

            using (var tif = new MagickImageCollection())
            {
                var page = 1;
                foreach (var image in images)
                {
                    Console.WriteLine("converting page " + page);
                    image.Format = MagickFormat.Tiff;
                    image.SetCompression(CompressionMethod.LZW);
                    tif.Add(image);

                    page++;
                }

                tif.Write(e.Name + ".tif", MagickFormat.Tiff);

            }
        }

But when looking at the resulting file with IrfanView, it tells me that the file is uncompressed - a 7 page PDF with 170 KB size is inflated to a 250 MB TIF file with 300 dpi density settings.

I thought "image.setCompression(...)" should do the trick... Apparently, this seems to be a recurring topic :-)

Using Magick.NET Version 9.1.2

**Comment by dlemstra** [comment]
Can you share your input file? The initial size does not mean anything because the image is a vector image and when it is read with ImageMagick it will be changed to a raster image.

**Reply by Tom-EllisEVENTS** [reply]
See attached ZIP file with the PDF and TIF file.

But I don't think the point is the actual file, because if I save the tiff file from IrfanView with LZW compression, it shrinks to a few 100 kb.

It seems like the SetCompression call is just ineffective.

While debugging, one thing caught my attention:

The compression attribute of each image in the "tif" collection is set to "LZW", but not _settings.compression, which is "undefined"

![2022-02-15 10_08_43-FileSystemWatcher (Debugging) - Microsoft Visual Studio](https://user-images.githubusercontent.com/6583210/154029773-8ff501f3-4ab3-4913-85f4-c9ff13e2ebad.png)

![2022-02-15 10_09_00-FileSystemWatcher (Debugging) - Microsoft Visual Studio](https://user-images.githubusercontent.com/6583210/154029851-b721aa3c-908c-46e1-8245-9e6d49edbafa.png)

[Lorem ipsum dolor sit amet.pdf.zip](https://github.com/dlemstra/Magick.NET/files/8067621/Lorem.ipsum.dolor.sit.amet.pdf.zip)

**Comment by dlemstra** [comment, marked as answer]
You should not call `SetCompression` but set the compression of the first image in the collection instead.

```C#
using (var images = new MagickImageCollection())
{
    images.Read("Lorem ipsum dolor sit amet.pdf", new MagickReadSettings
    {
        Density = new Density(150),
    });

    images[0].Settings.Compression = CompressionMethod.LZW;

    images.Write("test.tif");
}
```

**Reply by Tom-EllisEVENTS** [reply]
It works, but does this make sense...?

What is the reason that SetCompression() does not set Settings.Compression as well?

This is not very intuitive...

**Reply by dlemstra** [reply]
I agree that it isn't and that is why that method has the following comment:

> Gets the compression of the image. This method should only be used when the encoder uses the compression of the image. For most usecases Setting.Compression should be used instead.
